server/src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -893,13 +915,63 @@ public void setContentAccessMode(String contentAccessMode) {
 
     public void setAddOns(Set<String> addOns) {
         this.addOns = addOns;
+        this.updateRHCloudProfileModified();
     }
 
     public void addAddOn(String addOn) {
         this.addOns.add(addOn);
+        this.updateRHCloudProfileModified();
     }
 
     public void removeAddOn(String addOn) {
         this.addOns.remove(addOn);
+        this.updateRHCloudProfileModified();
+    }
+
+    public Date getRHCloudProfileModified() {
+        return this.rhCloudProfileModified;
+    }
+
+    public void setRHCloudProfileModified(Date rhCloudProfileModified) {
+        this.rhCloudProfileModified = rhCloudProfileModified;
+    }
+
+    public void updateRHCloudProfileModified() {
+        this.rhCloudProfileModified = new Date();
+    }
+
+    /**
+     * Check if the consumers facts have changed and contains the cloud profile facts.
+     * It returns true if incoming facts are cloud profile facts
+     *
+     * @param incomingFacts incoming facts
+     * @return a boolean
+     */
+    public boolean checkForCloudProfileFacts(Map<String, String> incomingFacts) {
+
+        if (incomingFacts == null) {
+            return false;
+        }
+
+        if (this.facts != null && incomingFacts != null && this.facts.equals(incomingFacts)) {
```

Comment:
we already know `incomingFacts` is not null here, since if it was we would have returned false, so there is no need for this check

Reply:
done.